### PR TITLE
feat: Update Electron to v6

### PR DESCRIPTION
**Pull request by ashchan**
* `dialog` APIs are promise based now, need to update usage.
* `ipcMain`'s `on` receives `IpcMainEvent`, not `Event`.
* Others?

## Needs Confirmation or Fix

* spectron hasn't released any update for electron v6 yet, e2e tests are behaving wrong
* sqlite3 native module build on Windows
* Others?

## Warning

Due to Electron's API change (`dialog` etc), there might be some changes our tests don't cover.  It's ok to leave them as holes and fix ASAP.

### CODE REVIEW

**Comment by ashchan** [comment]
Hold until next week to see if node-sqlite3 catches up.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -96,7 +96,7 @@ jobs:
         yarn global add lerna
         yarn bootstrap
       name: Bootstrap
-    - script: echo Disabling e2e test til spectron has update for electron v6 # yarn test:e2e
```
Comment (ashchan) [inline azure-pipelines.yml]: Errata: the commit message talked about electron 4.0.0. It should be 6.0.0 instead.

**Comment by Keith-CY** [comment]
Please rebase on the develop

**Comment by ashchan** [comment]
> Please rebase on the develop

Rebased already. Checking if I'm missing something locally.

**Comment by classicalliu** [comment]
It seems so strange that many line already in develop.
So it's hard to known what you really changed this PR from so large a change set.

**Comment by ashchan** [comment]
> It seems so strange that many line already in develop.
> So it's hard to known what you really changed this PR from so large a change set.

Rebased commits were pushed. Should be much smaller now. Sorry for the confusion.

**Pull request merged**